taxi/views.py

```diff
@@ -84,6 +86,39 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
     paginate_by = 5
 
 
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Driver
+    form_class = DriverCreateForm
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverLicenseUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+
+
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+@login_required
```

Comment:
Try to include this logic in CarDetailView - override post method to add/remove driver to car through related fields. You will also need to adjust your form on a template to pass car_id

Reply:
You can put logic from this view into post method of CarDetailView. We don't need to create extra view and endpoint to handle adding/deleting driver.
So the implementation will look something like this:

1. Add an empty form on car detail template that will send value 'Add' or 'Delete'
2. This form will send a request to CarDetailView (so no need to make extra endpoints) in a post method
3. You override post method in CarDetailView and check values that you received in if condition (you basically slightly modify code in your function view below and put in post method).